### PR TITLE
Changing the structure of the project to have the namespace and collections be parallel not nested

### LINKED ISSUE #432: Refactor our mediaflux Namespace/Collection hierarchy to match RC/Chuck's
As of right now, our local Mediaflux instance creates Collections (projects) to be children of Namespaces. They should instead have a 'cousin' relationship as opposed to a parent-child relationship.

Our structure will be as follows (notice the `/td-demo-001NS` cousin for namespaces):

```
/td-demo-001 (namespace)
/td-demo-001/project1 (collection)
/td-demo-001/project1 (collection)
/td-demo-001NS (namespace)
/td-demo-001NS/project1NS (namespace)
/td-demo-001NS/project2NS (namespace)
```

Notice that this mimics the structure of Mediaflux in production but this is **not** the structure used for `projectg` in the staging environment (`td-meta1`). 

We have code in our ProjectMediaflux class to create the rootnamespace that will need to be updated: https://github.com/pulibrary/tiger-data-app/blob/4ea19f3ffdb4a76a2587387fea3a20e9e67ca20a/app/models/project_mediaflux.rb#L72 

And also the code that sets the "project_namespace" on the same class: https://github.com/pulibrary/tiger-data-app/blob/4ea19f3ffdb4a76a2587387fea3a20e9e67ca20a/app/models/project_mediaflux.rb#L12

Acceptance Criteria
- [ ] The Collection for a project is created at within td-demo-001 and Namespace for a project is created under td-demo-001NS

### PR DESCRIPTION
**Pull request by carolyncole**
fixes #432

Also adds some better error handling for project creation

Screenshot 2024-02-16 at 9 49 00 AM

### CODE REVIEW

File: app/models/project_mediaflux.rb [hunk 1]
```diff
@@ -9,12 +9,17 @@ def self.create!(project:, session_id:, xml_namespace: nil)
     # Create a namespace for the project
     # The namespace is directly under our root namespace
     project_name = safe_name(project.directory)
-    project_namespace = "#{Rails.configuration.mediaflux['api_root_ns']}/#{project_name}-ns"
-    Mediaflux::Http::NamespaceCreateRequest.new(namespace: project_namespace, description: "Namespace for project #{project.title}", store: store_name, session_token: session_id).resolve
-    # Create a collection asset under the new namespace and set its metadata
+    project_namespace = "#{Rails.configuration.mediaflux['api_root_ns']}/#{project_name}NS"
+    namespace = Mediaflux::Http::NamespaceCreateRequest.new(namespace: project_namespace, description: "Namespace for project #{project.title}", store: store_name, session_token: session_id)
+    if namespace.error?
+      raise "Can not create the namespace #{namespace.response_error}"
```
Comment (hectorcorrea) [inline app/models/project_mediaflux.rb:15]: 👍🏼

File: app/models/mediaflux/http/namespace_create_request.rb [hunk 2]
```diff
@@ -27,7 +27,12 @@ def self.service
         def build_http_request_body(name:)
           super do |xml|
             xml.args do
-              xml.namespace namespace if namespace.present?
+              if namespace.present?
+                xml.namespace do
+                  xml.parent.set_attribute("all", true)
```
Comment (hectorcorrea) [inline app/models/mediaflux/http/namespace_create_request.rb:32]: what does the `all=true' do?

**Review by hectorcorrea — APPROVED** [review]

File: app/models/mediaflux/http/namespace_create_request.rb (hunk 2 above)
Comment (carolyncole) [reply app/models/mediaflux/http/namespace_create_request.rb:32]: It makes the parent namespaces if they do not exists.  It essentially allows the code to run on an empty docker instance as long as the schema has been created.

Comment (hectorcorrea) [reply app/models/mediaflux/http/namespace_create_request.rb:32]: Great, I didn't know we could do this. It makes the code so much flexible!

**Pull request merged**